go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -877,3 +899,31 @@ func (bsnc *BaseSecondaryNetworkController) requireDHCP(pod *corev1.Pod) bool {
 		bsnc.IsPrimaryNetwork() &&
 		bsnc.TopologyType() == types.Layer2Topology
 }
+
+func (bsnc *BaseSecondaryNetworkController) createVMSourcePodLSPOpsForLiveMigration(
+	kubevirtLiveMigrationStatus *kubevirt.LiveMigrationStatus,
+	nadName string, ops []ovsdb.Operation) ([]ovsdb.Operation, error) {
+	if kubevirtLiveMigrationStatus.State != kubevirt.LiveMigrationTargetDomainReady {
+		return ops, nil
+	}
+	// closing the sourcePod lsp to ensure traffic goes to the now ready targetPod.
+	srcPod := kubevirtLiveMigrationStatus.SourcePod
+	sourcePodLspName := bsnc.GetLogicalPortName(srcPod, nadName)
+	sourcePodLsp := &nbdb.LogicalSwitchPort{Name: sourcePodLspName}
+	sourcePodLsp, err := libovsdbops.GetLogicalSwitchPort(bsnc.nbClient, sourcePodLsp)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch switch port for node %s: %w", sourcePodLspName, err)
+	}
+
+	sourcePodLsp.Enabled = ptr.To(false)
+	switchName, err := bsnc.getExpectedSwitchName(srcPod)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch switch name for pod %s: %w", srcPod.Name, err)
+	}
+	ops, err = libovsdbops.CreateOrUpdateLogicalSwitchPortsOnSwitchOps(bsnc.nbClient, ops, &nbdb.LogicalSwitch{Name: switchName}, sourcePodLsp)
+	if err != nil {
+		return nil, fmt.Errorf("failed updating logical switch port %+v on switch %s: %w", *sourcePodLsp, switchName, err)
+	}
```

Comment:
Can we encapsulate this so it can be resued at the failed migration, the logic is the same

```golang
func (bsnc* ) setPodLogicalSwitchPortEnabledField(pod, enabled bool) (error, ops) {
	lsp  := &nbdb.LogicalSwitchPort{Name: bsnc.GetLogicalPortName(pod, nadName)}
	lsp, err := libovsdbops.GetLogicalSwitchPort(bsnc.nbClient, lsp)
	if err != nil {
		return nil, fmt.Errorf("failed to fetch switch port for node %s: %w", lsp.Name, err)
	}

	lsp.Enabled = ptr.To(enabled)
	switchName, err := bsnc.getExpectedSwitchName(pod)
	if err != nil {
		return nil, fmt.Errorf("failed to fetch switch name for pod %s: %w", pod.Name, err)
	}
	ops, err = libovsdbops.CreateOrUpdateLogicalSwitchPortsOnSwitchOps(bsnc.nbClient, ops, &nbdb.LogicalSwitch{Name: switchName}, lsp)
	if err != nil {
		return nil, fmt.Errorf("failed updating logical switch port %+v on switch %s: %w", *lsp, switchName, err)
	}
}

```

Then this will be used like 

```golang
func (bsnc *BaseSecondaryNetworkController) createVMSourcePodLSPOpsForLiveMigration(
	kubevirtLiveMigrationStatus *kubevirt.LiveMigrationStatus,
	nadName string, ops []ovsdb.Operation) ([]ovsdb.Operation, error) {
	if kubevirtLiveMigrationStatus.State != kubevirt.LiveMigrationTargetDomainReady {
		return ops, nil
	}
       return bsnc.setPodLogicalSwitchPortEnabledField(pod, false)
```

Reply:
DONE